<?xml version="1.0"?>

<launch>
  <!-- turtlebot -->
  <include file="$(find turtlebot_bringup)/launch/minimal.launch" />
  <include file="$(find turtlebot_navigation)/launch/includes/gmapping/gmapping.launch.xml" />
  <node pkg="robot_pose_publisher" type="robot_pose_publisher" name="robot_pose_publisher"/>

  <include file="$(find turtlebot_navigation)/launch/includes/move_base.launch.xml" />

  <!-- laser -->
  <node  pkg="urg_node" type="urg_node" name="urg_node" >
    <param name="angle_min" value="-1.6"/>
    <param name="angle_max" value="1.6"/>
  </node>
 
  <node pkg="tf" type="static_transform_publisher" name="laser_in_base_link"
    args="0.03 0.0 0.0 0.0 0.0 0.0 /base_link /laser 80" />

  <!-- joystick -->
  <node pkg="joy" type="joy_node" name="joy_node">
    <param name="joy_node/dev" value="/dev/input/js0"/>
  </node>
  <include file="$(find turtlebot_teleop)/launch/xbox360_teleop.launch"/>

  <!-- torob -->
  <node pkg="torob" type="move" name="move">
      <param name="robot_radius" value="0.2"/>
      <param name="perception_distance" value="1.0"/>
      <param name="goal_frame_id" value="odom"/>
      <param name="init_goal_x" value="2.0" />
      <param name="init_goal_y" value="-0.5" />
      <param name="linear_speed" value="0.2" />
      <param name="angular_speed" value="1.2" />
      <param name="cmd_topic" value="limbes" />
  </node>

  <node pkg="torob" type="map_to_vmap" name="torob_vmap" />

  <node pkg="torob" type="viewer" name="scan_vmap_view" output="screen">
      <param name="vmap_topic" value="/scan_vmap"/>
      <param name="scan_topic" value="/scan"/>
      <param name="scale" value="240.0" />
  </node>

  <!-- Web -->
  <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch"/>
</launch>
